src/Purebred.hs

```diff
@@ -107,6 +107,21 @@ processConfig =
   >=> confEditor id
   >=> (confFileBrowserView . fbHomePath) id
 
+
+randomChar :: RandomGen g => (Char, g) -> (Char, g)
+randomChar (_, g) = randomElement (['0'..'9'] <> ['a'..'z'] <> ['A'..'Z']) g
+
+randomElement :: RandomGen g => [a] -> g -> (a, g)
+randomElement xs g = let (i, g') = randomR (0, length xs - 1) g
+                     in (xs !! i, g')
+
+-- | Note: iterate returns a list including the first random character. That
+-- first random character is most often not Boundary compliant, so invoke
+-- iterate with the seed value which returns only boundary compliant characters.
+genBoundary :: RandomGen g => g -> String
+genBoundary g = let seed = randomChar (random g)
+                in fst <$> iterate randomChar seed
+
```

Comment:
This whole implementation seems really janky to me.  I'd propose one of two alternatives.

First alternative: define a newtype and instance `Random` for it, then use `randoms`.

Second alternative (simpler):
```haskell
genBoundary :: RandomGen g => g -> String
genBoundary = filter isBoundaryChar . randomRs (minimum boundaryChars, maximum boundaryChars)
  where
  boundaryChars ['0'..'9'] <> ['a'..'z'] <> ['A'..'Z']
  isBoundaryChar = (`elem` boundaryChars)
```

We use `randomRs` as an efficient hack so we don't generate chars outside the range that contains all valid boundary chars.

Reply:
I thought the range has to be something like `['a'..'z']`. I also thought why not use:
```Haskell
filter isBoundaryChar . randoms
```
`randoms` would produce the characters and we simply filter what is useful. But boy was that slow. Suppose it needs to be in the given range ...

Reply:
Ouch 😱